src/videouplinkbandwidthpolicy/NScaleVideoUplinkBandwidthPolicy.ts

```diff
@@ -99,4 +153,54 @@ export default class NScaleVideoUplinkBandwidthPolicy implements VideoUplinkBand
   setHasBandwidthPriority(hasBandwidthPriority: boolean): void {
     this.hasBandwidthPriority = hasBandwidthPriority;
   }
+
+  setTransceiverController(transceiverController: TransceiverController): void {
+    this.transceiverController = transceiverController;
+  }
+
+  async updateTransceiverController(): Promise<void> {
+    const settings = this.getStreamCaptureSetting();
+    if (!settings) {
+      return;
+    }
+
+    const encodingParams: RTCRtpEncodingParameters = this.calculateEncodingParameters(settings);
+    if (
+      !this.isEncodingParameterEqual(
+        encodingParams,
+        this.encodingParamMap.get(NScaleVideoUplinkBandwidthPolicy.encondingMapKey)
+      )
+    ) {
+      this.encodingParamMap.set(NScaleVideoUplinkBandwidthPolicy.encondingMapKey, encodingParams);
+      this.transceiverController.setEncodingParameters(this.encodingParamMap);
+    }
+  }
+
+  private isEncodingParameterEqual(
+    encoding1: RTCRtpEncodingParameters,
+    encoding2: RTCRtpEncodingParameters
+  ): boolean {
+    return (
+      encoding1.maxBitrate === encoding2.maxBitrate &&
+      encoding1.scaleResolutionDownBy === encoding2.scaleResolutionDownBy
```

Comment:
What about all of the other fields? If you're not going to compare them, probably don't use `equal` as the name — perhaps `encodingParametersHaveSameBitrateAndScaling`.

Reply:
Updated to use JSON.stringify to compare the 2 params.

Reply:
Yikes, no, don't do that. Not only is it extremely expensive, but the string representations of two objects aren't guaranteed to be the same.

Reply:
This is how it is done in the current default simulcast policy. I also thought it was expensive so only compare the 2 properties that we use here.